tests/suite/cli_self_upd.rs

```diff
@@ -350,12 +356,17 @@ async fn update_overwrites_programs_display_version() {
         .await;
 
     USER_RUSTUP_VERSION
-        .set_value(Some(PLACEHOLDER_VERSION))
+        .set_value(Some((Type::String, PLACEHOLDER_VERSION.as_bytes())))
         .unwrap();
     cx.config.expect_ok(&["rustup", "self", "update"]).await;
+    let version = OsString::from(version)
+        .encode_wide()
+        .flat_map(|v| vec![v as u8, (v >> 8) as u8])
+        .chain([0, 0])
+        .collect::<Vec<u8>>();
     assert_eq!(
-        USER_RUSTUP_VERSION.get_value::<String>().unwrap().unwrap(),
-        version,
+        USER_RUSTUP_VERSION.get_value().unwrap().unwrap().as_ref(),
```

Comment:
It would also be helpful to `assert_eq` the `Type` of the `Value`s in your tests as the original code did. Same for the other test file that you've modified.